mlir/include/mlir/Dialect/OpenMP/OpenMPOps.td

```diff
@@ -618,6 +618,47 @@ def AtomicUpdateOp : OpenMP_Op<"atomic.update"> {
   let verifier = [{ return verifyAtomicUpdateOp(*this); }];
 }
 
+//===----------------------------------------------------------------------===//
+// [5.1] 2.21.2 threadprivate Directive
+//===----------------------------------------------------------------------===//
+
+def ThreadprivateOp : OpenMP_Op<"threadprivate"> {
+  let summary = "threadprivate directive";
+  let description = [{
+    The threadprivate directive specifies that variables are replicated, with
+    each thread having its own copy.
+
+    Inside and outside the parallel region, the addressing of symbols are
+    created strictly followed by the ThreadprivateOp operation since they have
+    different meanings. The addressing of symbol outside the parallel region
+    gets its value in master thread, while the addressing of symbol inside the
+    parallel region gets the value in encountering thread. They should be
```

Comment:
Should we distinguish these two meanings with separate MLIR operations?

Reply:
From the dumped MLIR, it has this semantics since one is outside parallel op region, the other one is inside parallel op region. As the description mentioned, this operation is attached to the addressing of symbol to represent the threadprivate attribute. It seems ok to represent these meanings using the current operation def.